### PR TITLE
Remove non-inclusive wording

### PR DESCRIPTION
**Pull request by ormergi**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

/kind cleanup

> /kind feature
> /kind design
> /kind flake
> /kind failing
> /kind documentation
> /kind regression
> /kind example

**What this PR does / why we need it**:
Implement https://github.com/openperouter/openperouter/blob/main/enhancements/api-improvements.md#remove-non-inclusive-terminology - removing non-inclusive wording

**Special notes for your reviewer**:
The PR may seem large as it touch lots of files, please note most changes are repetitive.

**Release note**:

```release-note
API: OpenPERouter CRD:  `spec.tolerateMaster` and `spec.runOnMaster` replaced with Kubernetes native scheduling primitives `nodeSelector` and `tolerations`, `spec.affinity` now affect all OpenPERouter pods.
```

**AI Guidelines Acknowledgment**:

- [x] I have reviewed all changes in this PR, including any AI-generated content, and I take full responsibility for its accuracy and correctness.

## Summary by CodeRabbit

* **New Features**
  * Added node selectors, affinity rules, and tolerations for all OpenPERouter components.
  * Added default tolerations for master and control-plane taints.
  * Exposed pod scheduling options through the OpenPERouter configuration schema.

* **Breaking Changes**
  * Removed the `runOnMaster` and `tolerateMaster` settings.

* **Documentation**
  * Updated configuration guidance and networking terminology for clarity.

### CODE REVIEW

File: internal/hostnetwork/vni_test.go [hunk 1]
```diff
@@ -291,7 +291,7 @@ var _ = Describe("L3 VNI configuration", func() {
 	It("should leave veth MTU at default when no underlay interface is configured", func() {
 		// No fake underlay is set up here, so findUnderlayMTU returns 0
 		// and setVethMTUForTunnelOverhead must leave the veth MTU untouched. The
-		// host-side veth is not enslaved to any bridge in the L3 path
+		// host-side veth is not attached to any bridge in the L3 path
 		// (it is only attached to a VRF in the target ns), so the host
 		// leg's MTU reflects only what the code under test set.
```
[the remark this answers is not in the document]
Comment (ormergi) [reply internal/hostnetwork/vni_test.go]: Done

File: operator/internal/helm/chart.go [hunk 2]
```diff
@@ -122,6 +123,22 @@ func patchChartValues(envConfig envconfig.EnvConfig, crdConfig *operatorapi.Open
...
+	// CRD, so an unset field falls back to the chart's own default (e.g., the
+	// default tolerations for control-plane/master nodes).
+	// The typed API structs are converted to plain map/slice values via JSON, so
+	// the chart's values.schema.json (which only knows generic JSON types) can
+	// validate them.
+	if len(crdConfig.Spec.NodeSelector) > 0 {
+		openperouterValues["nodeSelector"] = toJSONValue(crdConfig.Spec.NodeSelector)
+	}
+	if len(crdConfig.Spec.Tolerations) > 0 {
+		openperouterValues["tolerations"] = toJSONValue(crdConfig.Spec.Tolerations)
```
[the remark this answers is not in the document]
Comment (ormergi) [reply operator/internal/helm/chart.go]: Done

File: operator/internal/helm/helm_test.go [hunk 3]
```diff
@@ -311,25 +312,72 @@ func TestParseChartWithGroutDisabled(t *testing.T) {
...
+		expectAffinity        *v1.Affinity
 	}{
 		{
-			name:              "runOnMaster enabled",
-			runOnMaster:       true,
-			expectTolerations: true,
+			name:                  "unset scheduling fields keep the chart default control-plane tolerations",
+			spec:                  operatorapi.OpenPERouterSpec{LogLevel: new(operatorapi.LogLevelInfo)},
+			expectMasterTolerated: true,
 		},
```
[the remark this answers is not in the document]
Comment (ormergi) [reply operator/internal/helm/helm_test.go]: Done

File: operator/internal/helm/helm_test.go [hunk 4]
```diff
@@ -311,77 +312,112 @@ func TestParseChartWithGroutDisabled(t *testing.T) {
...
-
 				if objKind == deploymentKind && obj.GetName() == nodemarkerDeploymentName {
 					deployment := appsv1.Deployment{}
 					err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), &deployment)
 					g.Expect(err).ToNot(HaveOccurred())
-					g.Expect(hasMasterToleration(deployment.Spec.Template.Spec.Tolerations)).To(Equal(tt.expectTolerations),
-						fmt.Sprintf("Deployment %s should have master toleration=%v", obj.GetName(), tt.expectTolerations))
+					podSpecs = append(podSpecs, deployment.Spec.Template.Spec)
+				}
+			}
```
[the remark this answers is not in the document]
Comment (ormergi) [reply operator/internal/helm/helm_test.go:415]: Done

**Review by maiqueb — COMMENTED** [review]
Don't we have any examples / website data to update with nodeSelectors / tolerations ? 

Other than that, the only thing I wonder is if we shouldn't split this PR in two:
- API changes (e.g. tolerations / nodeSelectors / ...)
- operator / helm charts

File: charts/openperouter/templates/controller.yaml [hunk 5]
```diff
@@ -143,17 +143,15 @@ spec:
...
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- with .Values.openperouter.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.openperouter.tolerations }}
-        {{- toYaml . | nindent 6 }}
+      tolerations:
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline charts/openperouter/templates/controller.yaml]: `tolerations:` is now emitted unconditionally, so an empty/null list renders `[]` at the *same* indentation as the key, which is invalid YAML:

```
$ helm template charts/openperouter --set-json 'openperouter.tolerations=[]'
Error: YAML parse error on openperouter/templates/controller.yaml: error converting YAML to JSON: yaml: line 105: could not find expected ':'
```

Rendered:
```yaml
      tolerations:
      []
      containers:
```

This is the migration path the PR prescribes — users who previously set `runOnMaster: false` must now set `tolerations: []` to opt out of control-plane scheduling, and that input hard-fails install/upgrade. Same pattern in `router.yaml`, `nodemarker.yaml`, `hostbridge.yaml` and the four `operator/bindata/...` copies.

Suggested fix (all 8 files):

```yaml
      {{- with .Values.openperouter.tolerations }}
      tolerations:
        {{- toYaml . | nindent 6 }}
      {{- end }}
```

File: charts/openperouter/templates/router.yaml [hunk 6]
```diff
@@ -350,17 +350,15 @@ spec:
...
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- with .Values.openperouter.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.openperouter.tolerations }}
-        {{- toYaml . | nindent 6 }}
+      tolerations:
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline charts/openperouter/templates/router.yaml]: Same unconditional `tolerations:` emission as in `controller.yaml` — `tolerations: []` renders invalid YAML. Needs the `{{- with }}` guard.

File: charts/openperouter/templates/hostbridge.yaml [hunk 7]
```diff
@@ -77,17 +77,15 @@ spec:
...
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- with .Values.openperouter.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.openperouter.tolerations }}
-        {{- toYaml . | nindent 6 }}
+      tolerations:
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline charts/openperouter/templates/hostbridge.yaml]: Same unconditional `tolerations:` emission as in `controller.yaml` — `tolerations: []` renders invalid YAML. Needs the `{{- with }}` guard.

File: charts/openperouter/templates/nodemarker.yaml [hunk 8]
```diff
@@ -34,14 +34,18 @@ spec:
...
+      {{- end }}
       tolerations:
-      {{- if .Values.openperouter.runOnMaster }}
-      - effect: NoSchedule
-        key: node-role.kubernetes.io/master
-        operator: Exists
-      - effect: NoSchedule
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline charts/openperouter/templates/nodemarker.yaml]: Same unconditional `tolerations:` emission as in `controller.yaml` — `tolerations: []` renders invalid YAML. Needs the `{{- with }}` guard.

File: charts/openperouter/templates/nodemarker.yaml [hunk 9]
```diff
@@ -34,14 +34,18 @@ spec:
...
+      {{- with .Values.openperouter.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (qinqon) [inline charts/openperouter/templates/nodemarker.yaml:48]: Behavior change: nodemarker previously had no `priorityClassName` block. This looks intentional (consistency with the other workloads) but isn't mentioned in the release note, which only covers the scheduling-primitive replacement.

File: operator/bindata/deployment/openperouter/templates/controller.yaml [hunk 10]
```diff
@@ -143,17 +143,15 @@ spec:
...
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- with .Values.openperouter.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.openperouter.tolerations }}
-        {{- toYaml . | nindent 6 }}
+      tolerations:
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline operator/bindata/deployment/openperouter/templates/controller.yaml]: Same unconditional `tolerations:` emission — `tolerations: []` renders invalid YAML. The operator hits this path too: `patchChartValues` sets the value whenever `Spec.Tolerations != nil`, so an `OpenPERouter` CR with `tolerations: []` produces an empty list and reconciliation fails on manifest rendering.

File: operator/bindata/deployment/openperouter/templates/router.yaml [hunk 11]
```diff
@@ -350,17 +350,15 @@ spec:
...
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- with .Values.openperouter.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.openperouter.tolerations }}
-        {{- toYaml . | nindent 6 }}
+      tolerations:
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline operator/bindata/deployment/openperouter/templates/router.yaml]: Same unconditional `tolerations:` emission — needs the `{{- with }}` guard.

File: operator/bindata/deployment/openperouter/templates/hostbridge.yaml [hunk 12]
```diff
@@ -77,17 +77,15 @@ spec:
...
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- with .Values.openperouter.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.openperouter.tolerations }}
-        {{- toYaml . | nindent 6 }}
+      tolerations:
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline operator/bindata/deployment/openperouter/templates/hostbridge.yaml]: Same unconditional `tolerations:` emission — needs the `{{- with }}` guard.

File: operator/bindata/deployment/openperouter/templates/nodemarker.yaml [hunk 13]
```diff
@@ -34,14 +34,18 @@ spec:
...
+      {{- end }}
       tolerations:
-      {{- if .Values.openperouter.runOnMaster }}
-      - effect: NoSchedule
-        key: node-role.kubernetes.io/master
-        operator: Exists
-      - effect: NoSchedule
-        key: node-role.kubernetes.io/control-plane
-        operator: Exists
+      {{- toYaml .Values.openperouter.tolerations | nindent 6 }}
```
Comment (qinqon) [inline operator/bindata/deployment/openperouter/templates/nodemarker.yaml]: Same unconditional `tolerations:` emission — needs the `{{- with }}` guard.

File: operator/internal/helm/chart.go [hunk 14]
```diff
@@ -122,6 +123,22 @@ func patchChartValues(envConfig envconfig.EnvConfig, crdConfig *operatorapi.Open
 		"cri": cri,
 	}
 
+	// Only set nodeSelector/tolerations/affinity when explicitly provided on the
+	// CRD, so an unset field falls back to the chart's own default (e.g., the
+	// default tolerations for control-plane/master nodes).
+	// The typed API structs are converted to plain map/slice values via JSON, so
+	// the chart's values.schema.json (which only knows generic JSON types) can
+	// validate them.
+	if len(crdConfig.Spec.NodeSelector) > 0 {
```
Comment (qinqon) [inline operator/internal/helm/chart.go]: `NodeSelector` is gated on `len(...) > 0` while `Tolerations`/`Affinity` are gated on `!= nil`. Behaviorally harmless today (the chart default for `nodeSelector` is `{}`), but the asymmetry means an explicitly-empty `nodeSelector` silently falls back to the chart default while an explicitly-empty `tolerations` does not. Worth aligning for predictability.

File: operator/internal/helm/chart.go [hunk 15]
```diff
@@ -160,3 +177,20 @@ func patchChartValues(envConfig envconfig.EnvConfig, crdConfig *operatorapi.Open
 		"enabled": false,
 	}
 }
+
+// toJSONValue converts a typed Go value (e.g. a Kubernetes API struct) into
+// the plain map[string]any/[]any/etc. representation that Helm's value
+// schema validation expects.
+func toJSONValue(v any) any {
+	data, err := json.Marshal(v)
+	if err != nil {
```
Comment (qinqon) [inline operator/internal/helm/chart.go:209]: Don't panic return the error.

`panic(err)` runs inside the operator's reconcile loop, so a marshal failure takes down the controller process rather than surfacing a reconcile error. The comment is right that failure is not expected with typed inputs, so this is low severity — but returning/propagating an error (or skipping the value) would be less disruptive than a crash.

File: operator/internal/helm/helm_test.go [hunk 16]
```diff
@@ -311,77 +313,113 @@ func TestParseChartWithGroutDisabled(t *testing.T) {
 	g.Expect(controllerFound).To(BeTrue())
```
Comment (qinqon) [inline operator/internal/helm/helm_test.go:313]: The test table covers the default and a non-empty tolerations override, but not the opt-out case. Adding a `Tolerations: []v1.Toleration{}` case would catch the empty-list rendering failure noted on the templates.

**Review by qinqon — CHANGES_REQUESTED** [review]

File: charts/openperouter/templates/controller.yaml (hunk 5 above)
Comment (maiqueb) [reply charts/openperouter/templates/controller.yaml]: @ormergi please add a test for this scenario - where you provide an empty slice of tolerations.

We want to ensure we never break it like this.

@qinqon good catch.

File: charts/openperouter/templates/nodemarker.yaml (hunk 9 above)
Comment (maiqueb) [reply charts/openperouter/templates/nodemarker.yaml:48]: I don't understand this comment. FWIW, I see it applied to a block of code about `affinity`

File: operator/internal/helm/chart.go (hunk 14 above)
Comment (maiqueb) [reply operator/internal/helm/chart.go]: +1 ; please align. I'd go w/ len > 0 - it seems to be broader.

File: charts/openperouter/templates/nodemarker.yaml (hunk 9 above)
Comment (qinqon) [reply charts/openperouter/templates/nodemarker.yaml:48]: My fault did it at wrong line, tt's about line 48 (`priorityClassName: {{ . | quote }}`), not the affinity block above it.

Comment (maiqueb) [reply charts/openperouter/templates/nodemarker.yaml:48]: okok, all good. 

I think we should not meddle with this - it's not asked for in the task.

We should instead open up an issue, and tackle it in a dedicated PR.

File: charts/openperouter/templates/controller.yaml (hunk 5 above)
Comment (ormergi) [reply charts/openperouter/templates/controller.yaml]: Good catch, thanks!
Done

File: charts/openperouter/templates/router.yaml (hunk 6 above)
Comment (ormergi) [reply charts/openperouter/templates/router.yaml]: Done

File: charts/openperouter/templates/hostbridge.yaml (hunk 7 above)
Comment (ormergi) [reply charts/openperouter/templates/hostbridge.yaml]: Done

File: charts/openperouter/templates/nodemarker.yaml (hunk 8 above)
Comment (ormergi) [reply charts/openperouter/templates/nodemarker.yaml]: Done

File: operator/bindata/deployment/openperouter/templates/controller.yaml (hunk 10 above)
Comment (ormergi) [reply operator/bindata/deployment/openperouter/templates/controller.yaml]: Done

File: operator/bindata/deployment/openperouter/templates/router.yaml (hunk 11 above)
Comment (ormergi) [reply operator/bindata/deployment/openperouter/templates/router.yaml]: Done

File: operator/bindata/deployment/openperouter/templates/hostbridge.yaml (hunk 12 above)
Comment (ormergi) [reply operator/bindata/deployment/openperouter/templates/hostbridge.yaml]: Done

File: operator/bindata/deployment/openperouter/templates/nodemarker.yaml (hunk 13 above)
Comment (ormergi) [reply operator/bindata/deployment/openperouter/templates/nodemarker.yaml]: Done

File: operator/internal/helm/chart.go (hunk 14 above)
Comment (ormergi) [reply operator/internal/helm/chart.go]: Done

File: operator/internal/helm/chart.go (hunk 15 above)
Comment (ormergi) [reply operator/internal/helm/chart.go:209]: I dont know why I though it affect tests only, my bad.
Done

File: operator/internal/helm/helm_test.go (hunk 16 above)
Comment (ormergi) [reply operator/internal/helm/helm_test.go:313]: Done

**Comment by ormergi** [comment]
[Changes](https://github.com/openperouter/openperouter/compare/4dc37ff75514ca33df3f4e56495b7b71e9569749..e0840cadb588e636b6166217d8bcc46721747824): resolve https://github.com/openperouter/openperouter/pull/649#pullrequestreview-4864980583, @qinqon @maiqueb  PTAL

**Review by maiqueb — CHANGES_REQUESTED** [review]
Thank you.

I was about to approve the PR when I realized you didn't drop the priority class changes from the PR. Please drop those, and open a follow up PR with it. (or track that via an issue).

**Comment by ormergi** [comment]
> Don't we have any examples / website data to update with nodeSelectors / tolerations ?

@maiqueb I didnt find any
In general these are k8s native APIs, should be straight forward. Also the added attributes have GODOC that can be viewd with `kubectl explain openperrouter`.

>Other than that, the only thing I wonder is if we shouldn't split this PR in >two:

>    API changes (e.g. tolerations / nodeSelectors / ...)
    operator / helm charts

If the PR is converging lets continue with it, if at some point you feel it diverging let me know and I will export helm related changes to another PR.

**Comment by ormergi** [comment]
> Thank you.

>I was about to approve the PR when I realized you didn't drop the priority class changes from the PR. Please drop those, and open a follow up PR with it. (or track that via an issue).

@maiqueb 
Done
FWIW touching priorityClassName is not intended, it mistakenly slipped in

File: charts/openperouter/templates/nodemarker.yaml (hunk 9 above)
Comment (ormergi) [reply charts/openperouter/templates/nodemarker.yaml:48]: FWIW touching priorityClassName was never my intention in this PR, it was  mistakenly slipped in.

**Comment by maiqueb** [comment]
> > Don't we have any examples / website data to update with nodeSelectors / tolerations ?
> 
> @maiqueb I didnt find any In general these are k8s native APIs, should be straight forward. Also the added attributes have GODOC that can be viewd with `kubectl explain openperrouter`.
> 
> > Other than that, the only thing I wonder is if we shouldn't split this PR in >two:
> 
> > API changes (e.g. tolerations / nodeSelectors / ...)
> > operator / helm charts
> 
> If the PR is converging lets continue with it, if at some point you feel it diverging let me know and I will export helm related changes to another PR.

The PR is converging, but, IIRC, we did meet a couple of weeks ago, and we've decided on providing smaller scoped PRs, with tackle a single thing. 

This is the opposite of that - hence the question.

**Review by maiqueb — APPROVED** [review]
Tks.

**Comment by ormergi** [comment]
>The PR is converging, but, IIRC, we did meet a couple of weeks ago, and we've decided on providing smaller scoped PRs, with tackle a single thing.

>This is the opposite of that - hence the question.

I get that yes, this PR may seem large because it touch lots of files, but the changes are repetitive spreading across various manifests because k8s scheduling primitives added exist in part of the pod spec. So natuarlly it would touch all files involve pod spec / template.
I added a comment for it in the PR description, to clear out the expectations for reviewers.

IMO this PR is fraction in terms of size and scope comparing to the configuration-reliance or the node-status API work, and recent features such as grout support or multus-replacement.
FWIW this is why I allowed to myself to post his PR as is.

**Comment by ormergi** [comment]
Posted issue for tracking followup work on closing gap we found for priorityClassName in node-marker manifest https://github.com/openperouter/openperouter/issues/666

**Comment by maiqueb** [comment]
> > The PR is converging, but, IIRC, we did meet a couple of weeks ago, and we've decided on providing smaller scoped PRs, with tackle a single thing.
> 
> > This is the opposite of that - hence the question.
> 
> I get that yes, this PR may seem large because it touch lots of files, but the changes are repetitive spreading across various manifests because k8s scheduling primitives added exist in part of the pod spec. So natuarlly it would touch all files involve pod spec / template. I added a comment for it in the PR description, to clear out the expectations for reviewers.
> 
> IMO this PR is fraction in terms of size and scope comparing to the configuration-reliance or the node-status API work, and recent features such as grout support or mulch-replacment. FWIW at lest this is why I allowed to myself to post his PR as is.

I simply heard that to **you** this is OK, but the others were not. 

We should have a clear line (which we have, if we adopt this workflow *everytime*), rather than force ourselves to make the call at every PR.

**Review by qinqon — APPROVED** [review]
LGTM

**Comment by maiqueb** [comment]
@ormergi please rebase.

**Comment by ormergi** [comment]
[Rebased](https://github.com/openperouter/openperouter/compare/aaf1ba3a5d0f26bb500f4e59446e4b2b83f0068a..f8ac92ab96e3ba27079f085b5d1e504940e92440)
